src/event.rs

```diff
@@ -74,9 +74,93 @@ pub enum Event<'a, T: 'static> {
     UserEvent(T),
 
     /// Emitted when the application has been suspended.
+    ///
+    /// # Portability
+    ///
+    /// Not all platforms support the notion of suspending applications, and there may be no
+    /// technical way to guarantee being able to emit a `Suspended` event if the OS has
+    /// no formal application lifecycle (currently only Android and iOS do). For this reason
+    /// Winit does not currently try to emit pseudo `Suspended` events before the application
+    /// quits on platforms without an application lifecycle.
+    ///
+    /// In case of corner cases with how platform-specific events are mapped to Winit `Suspended`
+    /// and `Resumed` events it's recommended that applications be able to gracefully handle redundant
+    /// (i.e. back to back) `Suspended` or `Resumed` events.
+    ///
+    /// Also see [`Resumed`][Self::Resumed] notes.
+    ///
+    /// # Android
+    ///
+    /// On Android specifically, the `Suspended` event relates to two things:
+    /// 1. The application is paused (according to the [Activity lifecycle][android_ls])
+    /// 2. Or the application has had it's associated SurfaceView destroyed (which is expected
+    /// to closely correlate with being paused)
+    ///
+    /// Applications that need to run on Android should assume their SurfaceView has been
+    /// destroyed, which indirectly invalidates any existing render surfaces that may have been
+    /// created outside of Winit (such as an `EGLSurface`, `VkSurfaceKHR` or [`wgpu::Surface`][wgpu_surface]).
+    ///
+    /// [wgpu_surface]: https://docs.rs/wgpu/latest/wgpu/struct.Surface.html
+    ///
+    /// After being `Suspended` on Android it's recommended to immediately drop all render
+    /// surfaces which may be re-created when the application is next `Resumed`
```

Comment:
```suggestion
    /// surfaces which may be re-created when the application is next `Resumed`.
```

Not sure about "recommended", it's pretty much required if no-one keeps the `NativeWindow` alive (`winit` won't) otherwise leading to UB?

Reply:
yeah I wrote "recommended" In the sense that you could maybe technically avoid doing anything that tries to interact with underlying surface view/`NativeWindow` and you could might get away with that but yeah I might be wrong about that. I'll say it's a "must".

Reply:
I also added the clarification that the render surfaces should be dropped before the the event callback is complete, since that may related to how the backend implementation synchronizes with Java:

```rust
    /// After being `Suspended` on Android applications must drop all render surfaces before
    /// the event callback completes, which may be re-created when the application is next `Resumed`
```

Reply:
Yeah I like that wording.

Not sure either, I don't think Vulkan/EGL "acquire" the nativewindow so it _might_ be UB - but it's not if the caller acquires (by means of cloning) a NativeWindow externally. Oh my, we don't need to document that case.

Reply:
The main takeaway - I guess - is that a render surface can only be created after Resumed, and that you must do that every time. A render surface (producer) _may_ stay valid after Suspended (re. acquire above) but it'd be useless if android gives you a new, different producer. Then users wonder why their screen stays black :)